runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -76,58 +78,56 @@ public int size() {
      * This class encapsulates these two longs into one Id and add rules on comparability.
      */
     public static class CorfuRecordId implements Comparable<CorfuRecordId> {
-        private final UUID id;
-        public CorfuRecordId(long ordering, long uniqueId) {
-            this.id = new UUID(ordering, uniqueId);
-        }
+        @Setter
+        @Getter
+        private long epoch;
 
-        /**
-         * @return Return only the unique part of the id without the ordering
-         */
-        public long getEntryId() {
-            return id.getLeastSignificantBits();
-        }
+        @Setter
+        @Getter
+        private long sequence;
 
-        /**
-         * @return Return only the ordering part of the entry without the id.
-         */
-        public long getOrdering() {
-            return id.getMostSignificantBits();
+        @Getter
+        private long entryId;
+
+        public CorfuRecordId(long epoch, long sequence, long entryId) {
+            this.epoch = epoch;
+            this.sequence = sequence;
+            this.entryId = entryId;
         }
 
         /**
          * It's NOT ok to compare two objects if their ordering metadata is dissimilar.
          * @param o object to compare against.
-         * @throws IllegalArgumentException if the two Ids are not comparable.
          * @return results of comparison.
          */
         @Override
         public int compareTo(CorfuRecordId o) {
-            if (this.id.getMostSignificantBits() == 0 && o.id.getMostSignificantBits() != 0) {
-                throw new IllegalArgumentException(
-                        "Incompatible CorfuRecordId comparison: ordering unavailable");
-            }
-            if (this.id.getMostSignificantBits() !=0 && o.id.getMostSignificantBits() == 0) {
-               throw new IllegalArgumentException(
-                       "Incompatible CorfuRecordId comparison: order of compared object unknown");
-            }
-            if (this.id.getLeastSignificantBits() == o.id.getLeastSignificantBits()) {
-                return 0;
-            }
-            return id.compareTo(o.id);
+            return Comparator.comparing(CorfuRecordId::getEpoch)
+                    .thenComparing(CorfuRecordId::getSequence)
+                    .thenComparing(CorfuRecordId::getEntryId)
+                    .compare(this, o);
         }
 
         /**
-         * It is ok to check equality of a CorfuRecordId with ordering data against one without.
-         * @param o object to compare against.
+         * It is NOT ok to check equality of a CorfuRecordId with ordering data against one without.
+         * @param otherObj object to compare against.
          * @return
          */
-        public boolean equals(CorfuRecordId o) {
-            return id.getLeastSignificantBits() == o.id.getLeastSignificantBits();
+        @Override
+        public boolean equals(Object otherObj) {
+            if (this == otherObj) return true;
```

Comment:
Actually instead of bolierplate code you should use @EqualsAndHashCode annotation from lombok. Can you make that change?

Reply:
done. thanks

Reply:
done. thanks